### PR TITLE
[plex] add ratingKey channels

### PR DESCRIPTION
**Pull request by ccutrer**
Useful for automations related to exactly what's playing, to avoid having to lookup based on title (which may not be unique). I use this to lookup the aspect ratio of the currently playing title in a database, and adjust my constant-image-height screen appropriately.

### CODE REVIEW

File: bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexPlayerHandler.java [hunk 1]
```diff
@@ -179,6 +182,14 @@ public synchronized void updateChannels() {
                 new StringType(String.valueOf(foundInSession ? currentSessionData.getEndTime() : "")));
         updateState(new ChannelUID(getThing().getUID(), CHANNEL_PLAYER_USER),
                 new StringType(String.valueOf(foundInSession ? currentSessionData.getUserTitle() : "")));
+        String parentRatingKey = currentSessionData.getParentRatingKey();
+        String grandparentRatingKey = currentSessionData.getGrandparentRatingKey();
```
Comment (mlobstein) [inline bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexPlayerHandler.java:186]:
```suggestion
        final String parentRatingKey = currentSessionData.getParentRatingKey();
        final String grandparentRatingKey = currentSessionData.getGrandparentRatingKey();
```

File: bundles/org.openhab.binding.plex/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -65,6 +68,25 @@
...
+		<description>The unique key in the Plex library identifying the parent (TV show season or album) of the media that is
+			playing</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="grandparentRatingKey" advanced="true">
+		<item-type>String</item-type>
+		<label>Grandparent Rating Key</label>
+		<description>The unique key in the Plex library identifying the grandparent (TV show) of the media that is playing</description>
+		<state readOnly="true"/>
+	</channel-type>
```
Comment (mlobstein) [inline bundles/org.openhab.binding.plex/src/main/resources/OH-INF/thing/thing-types.xml:89]: Upgrade instructions are needed to automatically add the new channels. See here for an example: https://github.com/openhab/openhab-addons/pull/14799/files#diff-f5712db66efe98f83c01f11166123dcdf146398d811ef519f2b2836a39f59fe3

**Review by mlobstein — CHANGES_REQUESTED** [review]

**Comment by mlobstein** [comment]
@aronbeurskens please review also

File: bundles/org.openhab.binding.plex/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (ccutrer) [reply bundles/org.openhab.binding.plex/src/main/resources/OH-INF/thing/thing-types.xml:92]: Done. And I tested both upgrading from a thing created before this change, and a thing created after this change.

**Review by aronbeurskens — APPROVED** [review]

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**